core/components/handler/pktStorage.go

```diff
@@ -145,6 +147,7 @@ func (e pktEntry) MarshalBinary() ([]byte, error) {
 	rw := readwriter.New(nil)
 	rw.Write(e.AppEUI)
 	rw.Write(e.DevEUI)
+	rw.Write(uint16(e.FPort))
```

Comment:
What do you mean with "currently not able to send any downlink data anyway"?
The current implementation just sets the downlink port to 1, which is working just fine.